contributors/design-proposals/controller-ref.md

```diff
@@ -1,101 +1,363 @@
+      Controllers should never be waiting for creation of an orphan because
+      anything they create should have a ControllerRef.
+  * *Delete* event
+    * Do *not* sync any controllers.
+      Controllers should never care about orphans disappearing.
+    * Do *not* update counters on `expectations`.
+      Controllers should never be waiting for deletion of an orphan because they
+      are not allowed to delete objects they don't own.
+  * *Update* event
+    * If labels changed, sync any controllers whose selectors match to give each
+      one a chance to adopt the object.
+
+## Default garbage collection policy
+
+Most controllers that implement ControllerRef should have a
+[`DefaultGarbageCollectionPolicy`](https://github.com/kubernetes/kubernetes/blob/dd22743b54f280f41e68f206449a13ca949aca4e/pkg/genericapiserver/registry/rest/delete.go#L43)
+of `rest.OrphanDependents`.
+
+This ensures that deleting only the controller, without specifying the optional
+`DeleteOptions.OrphanDependents` flag, remains a non-cascading delete.
+Otherwise, the behavior would change to server-side cascading deletion by
+default as soon as the controller manager is upgraded to a version that performs
+adoption by setting ControllerRefs.
+
+Example from [ReplicationController](https://github.com/kubernetes/kubernetes/blob/9ae2dfacf196ca7dbee798ee9c3e1663a5f39473/pkg/registry/core/replicationcontroller/strategy.go#L49):
+
+```go
+// DefaultGarbageCollectionPolicy returns Orphan because that was the default
+// behavior before the server-side garbage collection was implemented.
+func (rcStrategy) DefaultGarbageCollectionPolicy() rest.GarbageCollectionPolicy {
+	return rest.OrphanDependents
+}
+```
 
-# Considered alternatives
+## Controller-specific behavior
+
+This section lists considerations specific to a given controller.
+
+* **StatefulSet**
+
+  * A StatefulSet will not adopt any Pod whose name does not match the template
+    it uses to create new Pods: `{statefulset name}-{ordinal}`.
+    This is because Pods in a given StatefulSet form a "family" that may use pod
+    names (via their generated DNS entries) to coordinate among themselves.
+    Adopting Pods with the wrong names would violate StatefulSet's semantics.
+
+    Adoption is allowed when Pod names match, so it remains possible to orphan a
+    family of Pods (by deleting their StatefulSet without cascading) and then
+    create a new StatefulSet with the same name and selector to adopt them.
+
+* **CronJob**
+
+  * CronJob [does not use watches](https://github.com/kubernetes/kubernetes/blob/9ae2dfacf196ca7dbee798ee9c3e1663a5f39473/pkg/controller/cronjob/cronjob_controller.go#L20),
+    so [that section](#watches) doesn't apply.
+    Instead, all CronJobs are processed together upon every "sync".
+  * CronJob applies a `created-by` annotation to link Jobs to the CronJob that
+    created them.
+    If a ControllerRef is found, it should be used instead to determine this
```

Comment:
I'm just worried about the latter. I'm fine with the change itself, just need some time to let it soak in.